### PR TITLE
[NEW] Add share app

### LINKED ISSUE #1839: [NEW] Add share the app feature to share the Google Play link in other apps
## Description
Currently there is no way available to share the app. Add  feature to share the app by all possible ways available to phone.

## Devices and Versions
Your Rocket.Chat.Android version: 3.1.0 Build 2049 - 60b39dad-play

**Would you like to work on the issue?**
- Yes

### PR DESCRIPTION
**Pull request by GOVINDDIXIT**
@RocketChat/android

Closes #1839

##GIF
gif-181106_153551

### CODE REVIEW

File: app/src/main/java/chat/rocket/android/settings/ui/SettingsFragment.kt [hunk 1]
```diff
@@ -75,6 +77,16 @@ class SettingsFragment : Fragment(), SettingsView, AdapterView.OnItemClickListen
                     AboutFragment.newInstance()
                 }
             }
+            resources.getString(R.string.title_share_the_app) ->{
+                val shareIntent = Intent(Intent.ACTION_SEND)
+                shareIntent.type = "text/plain"
+                val shareBody = "Check This Out"
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/settings/ui/SettingsFragment.kt]: Could you also move this hardcoded string to all the strings.xml?

File: app/src/main/java/chat/rocket/android/settings/ui/SettingsFragment.kt [hunk 2]
```diff
@@ -75,6 +77,16 @@ class SettingsFragment : Fragment(), SettingsView, AdapterView.OnItemClickListen
...
+                val shareSub = "https://play.google.com/store/apps/details?id=chat.rocket.android"
+                shareIntent.putExtra(Intent.EXTRA_SUBJECT, shareBody)
+                shareIntent.putExtra(Intent.EXTRA_TEXT, shareSub)
+                startActivity(Intent.createChooser(shareIntent, "Share Using"))
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/settings/ui/SettingsFragment.kt]: Could you also move this hardcoded string to all the strings.xml?

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Comment by GOVINDDIXIT** [comment]
@filipedelimabrito pls review
Also I had corrected the [PR](https://github.com/RocketChat/Rocket.Chat.Android/pull/1837)

**Comment by GOVINDDIXIT** [comment]
updated

File: app/src/main/res/values-de/strings.xml [hunk 3]
```diff
@@ -300,6 +304,9 @@
     <string name="header_live_chats">Live Chats</string>
     <string name="header_unknown">Unbekannt</string>
 
+    <!--Links-->
+    <string name="play_store_link">"https://play.google.com/store/apps/details?id=chat.rocket.android"</string>
```
Comment (philipbrito) [inline app/src/main/res/values-de/strings.xml]: Could you please move this string to the `defaults.xml` file? Don't forget to remove the same string from the others strings.xml files as well. Thank you.

**Review by philipbrito — CHANGES_REQUESTED** [review]

**Comment by GOVINDDIXIT** [comment]
updated

**Comment by GOVINDDIXIT** [comment]
squashed the irrelevant commit now it is complete from my side. pls review

File: app/src/main/java/chat/rocket/android/settings/ui/SettingsFragment.kt [hunk 4]
```diff
@@ -1,6 +1,8 @@
 package chat.rocket.android.settings.ui
 
+import android.content.ActivityNotFoundException
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/settings/ui/SettingsFragment.kt]: Where is this import being used (the same for the `android.net.Uri`) ? 🤔

Comment (GOVINDDIXIT) [reply app/src/main/java/chat/rocket/android/settings/ui/SettingsFragment.kt]: Oh my bad, forgot to remove

**Review by philipbrito — APPROVED** [review]
Thank you so much for submitting this PR @GOVINDDIXIT !

**Comment by GOVINDDIXIT** [comment]
@filipedelimabrito Thanks for guidance, always ready to learn from you😊

**Comment by philipbrito** [comment]
It is always a pleasure @GOVINDDIXIT! Feel free to reach us anytime! All your PRs are welcome! 👊

**Pull request merged**